keps/sig-apps/3850-backoff-limits-per-index-for-indexed-jobs/README.md

```diff
@@ -0,0 +1,1273 @@
+reached.
+
+### User Stories (Optional)
+
+#### Story 1
+
+As a CI/CD platform administrator, I want to use Indexed Jobs to run
+suites of integration tests, one suite per index. A failure of one suite should
+not interrupt running of other suites. Additionally, I would like to be able
+to control the maximal number of retries per index.
+
+The following Job configuration could satisfy my use case:
+
+```yaml
+apiVersion: v1
+kind: Job
+spec:
+  parallelism: 10
+  completions: 10
+  completionMode: Indexed
+  runPolicy:
+    backoffLimitPerIndex: 1
+  template:
+    spec:
+      restartPolicy: Never
+      containers:
+      - name: job-container
+        image: job-image
+        command: ["./tests-runner"]
+```
+
+In this case, we run 10 indexes representing the test suites. We allow for one
+failure per index.
+
+#### Story 2
+
+As a CI/CD platform administrator from the [Story 1](#story-1) I want to be able
+to control the failures with the pod failure policy. In particular, I want
+to be able to use pod failure policy to avoid restarts of some indexes, based
+on exit codes.
+
+The following Job configuration could satisfy my use case:
+
+```yaml
+apiVersion: v1
+kind: Job
+spec:
+  parallelism: 10
+  completions: 10
+  completionMode: Indexed
+  runPolicy:
+    backoffLimitPerIndex: 1
+  template:
+    spec:
+      restartPolicy: Never
+      containers:
+      - name: job-container
+        image: job-image
+        command: ["./tests-runner"]
+  podFailurePolicy:
+    rules:
+    - action: FailIndex
+      onExitCodes:
+        operator: In
+        values: [42]
+```
+
+#### Story 3
+
+As a CI/CD platform administrator from the [Story 1](#story-1) I want to be able
+to fail the entire Job if the number of failed indexes exceeds 50%. I want to
+do this in order to cut down costs of running the tests in case of compilation
+issues that would result in all tests failing.
+
+The following Job configuration could satisfy my use case:
+
+```yaml
+apiVersion: v1
+kind: Job
+spec:
+  parallelism: 10
+  completions: 10
+  completionMode: Indexed
+  runPolicy:
+    backoffLimitPerIndex: 1
+    maxFailedIndexes: 5
+  template:
+    spec:
+      restartPolicy: Never
+      containers:
+      - name: job-container
+        image: job-image
+        command: ["./tests-runner"]
+```
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
+
+
+#### The Job object too big
+
+With the new field `.status.failedIndexes` the Job object can be significantly
+larger as every failed index is recorded in the field. For example, using the
+same format as for the `.status.completedIndexes` and allowing for completions
+equal to `10^6` the size of the field, in the worst-case scenario is 3.4MB, which
```

Comment:
Actually, the worst case is much bigger, as @mwielgus pointed out offline. I opened a separate issue for the existing functionality https://github.com/kubernetes/kubernetes/issues/118085

Reply:
Good point, updated the section.